### ISSUE TITLE
Accessing ethers in my hre extension breaks gas-reporter

### ISSUE THREAD

**Issue by Lopol2010** [cgewecke/hardhat-gas-reporter#89]
Hi, I have this in my hardhat config and this part `hre.ethers.ContractFactory` is breaking gas-reporter.
```js
extendEnvironment((hre) => {
  hre.ethers.ContractFactory
});
```

Steps to reproduce:
1.  Initially plugin will report only header, run `npx hardhat test` to see
2. Comment out  `hre.ethers.ContractFactory` in hardhat config
3. See full report generated `npx hardhat test`

Repo: https://github.com/Lopol2010/gas-reporter-issue

**Comment by cgewecke** [comment]
@Lopol2010 Thanks for these reproduction steps, will take a look.

**Comment by nkuba** [comment]
We experienced a similar problem. Really looking forward to a fix.

**Comment by cgewecke** [comment]
Any issues with this should be fixed in the latest version (2.0). The reporter now uses Hardhat's `extendProvider` plugin hook to wrap the provider and this should resolve any weird conflicts that previously existed. 

Closing for now but if anyone runs into this please ping. 

https://github.com/cgewecke/hardhat-gas-reporter/releases/tag/v2.0.0
